tools/lsp/util.rs

```diff
@@ -185,9 +195,16 @@ fn lookup_expression_context(mut n: SyntaxNode) -> Option<(syntax_nodes::Element
         }
         match n.kind() {
             SyntaxKind::Binding | SyntaxKind::TwoWayBinding | SyntaxKind::CallbackConnection => {
-                let prop_name = i_slint_compiler::parser::identifier_text(&n)?;
-                let element = syntax_nodes::Element::new(n.parent()?)?;
-                break (element, prop_name);
+                let parent = n.parent()?;
+                if parent.kind() == SyntaxKind::PropertyAnimation {
+                    let prop_name = i_slint_compiler::parser::identifier_text(&n)?;
+                    let element = syntax_nodes::Element::new(parent.parent()?)?;
+                    break (element, prop_name);
+                } else {
+                    let prop_name = i_slint_compiler::parser::identifier_text(&n)?;
+                    let element = syntax_nodes::Element::new(parent)?;
+                    break (element, prop_name);
+                }
```

Comment:
The reason why I changed here is because the parent of animate property is animate itself, so the original 

> let element = syntax_nodes::Element::new(n.parent()?)?;

will cause early return and no completion at all. So here we need to get `parent of parent` for current node `n` if `parent` itself is `PropertyAnimation`

Reply:
There are other reason why this may fail, for example in a state, or a transition  (for which, incidentally we do not support auto-completion yet either).

I think it would be worth to add information to the return type of this function to take some more context into account.

```rust
struct ExpressionContextInfo { parent_element: syntax_nodes::Element, property_name: String, is_animate: bool }
fn lookup_expression_context(mut n: SyntaxNode) -> Option<ExpressionContextInfo> {
```

then you could use is_animate in with_property_lookup_ctx to find out whether you are in such a context.  What do you think?
But this doesn't have to be done now and can be done in another followup PR if needed.

Reply:
Yes, I agree. Actually, I was also thinking if I shoud change the return type of `lookup_expression_context`, but that may expand the scope of this task so I chose to take the shortcut.

I'll make change in this PR.